src/dcd/server/autocomplete/complete.d

```diff
@@ -114,8 +114,15 @@ AutocompleteResponse dotCompletion(T)(T beforeTokens, const(Token)[] tokenArray,
 		// of at the end
 		auto t = beforeTokens[$ - 1];
 		if (cursorPosition - t.index >= 0 && cursorPosition - t.index <= t.text.length)
+		{
 			partial = t.text[0 .. cursorPosition - t.index];
-		significantTokenType = tok!"identifier";
+			// issue 442 - prevent `partial` to start in the middle of a MBC
+			// since later there's a non-nothrow call to `toUpper`
+			import std.utf : validate, UTFException;
+			try validate(partial);
+			catch (UTFException) partial = "";
```

Comment:
I think that phobos misses a nothrow UTF validator. Using `try` `catch` for this is really horrible.

Reply:
Agreed. I also miss a way to process strings in `nothrow` e.g. by replacing with the invalid UTF char - though there was a DIP once: https://wiki.dlang.org/DIP76

Reply:
FYI: I recently hacked a `toValidUTF` method for run.dlang.io, because of similar problems.
Though it's really ugly (performance-wise), but Phobos is really lacking in this regard :/

https://github.com/dlang-tour/core/pull/673

Reply:
> Though it's really ugly (performance-wise), 

Thinking about it - maybe we should use this approach in the `catch(UTFException)` case? silently setting `partial` to an empty string if there's just one invalid UTF symbol looks a bit like patching over the actual problem.

Reply:
The actual problem comes from the IDE in first place and in dcd we just avoid a crash.